src/eth_rpc/servers/eth_rpc.rs

```diff
@@ -77,7 +77,11 @@ impl<P: Provider + Send + Sync + 'static> EthApiServer for KakarotEthRpc<P> {
     }
 
     async fn chain_id(&self) -> Result<Option<U64>> {
-        Ok(Some(CHAIN_ID.into()))
+        let chain_id_as_field = self.kakarot_client.starknet_provider().chain_id().await.map_err(EthApiError::from)?;
```

Comment:
in Rust, it's idiomatic to shadow variable declarations, unlike Typescript, where you can't.

That is:
```rust
let chain_id = self.kakarot_client.starknet_provider().chain_id().await.map_err(EthApiError::from)?;

let chain_id = u64::try_from(chain_id).ok().unwrap;
```

Reply:
Another approach, depending on taste is to use scope:
https://doc.rust-lang.org/rust-by-example/variable_bindings/scope.html

```rust
let chain_id = {
let chain_id = self.kakarot_client.starknet_provider().chain_id().await.map_err(EthApiError::from)?;
let chain_id = u64::try_from(chain_id).ok().unwrap;
U64([chain_id_as_u64])
}
```

Reply:
thank you!